torch/testing/_internal/common_utils.py

```diff
@@ -1426,111 +1339,37 @@ def _is_dict(self, obj):
-                                     atol=atol, rtol=rtol, msg=msg,
-                                     exact_dtype=exact_dtype, exact_device=exact_device)
-                    self.assertEqual(x.q_per_channel_axis(), y.q_per_channel_axis(),
-                                     atol=atol, rtol=rtol, msg=msg,
-                                     exact_dtype=exact_dtype, exact_device=exact_device)
-
-                result, debug_msg_compare = self._compareTensors(x.int_repr().to(torch.int32),
-                                                                 y.int_repr().to(torch.int32),
-                                                                 atol=atol, rtol=rtol,
-                                                                 exact_dtype=exact_dtype,
-                                                                 exact_device=exact_device)
-
-                if not result:
-                    assert debug_msg_compare is not None
-                    debug_msg = "Quantized representations failed to compare as equal! " + debug_msg_compare
-                super().assertTrue(result, msg=self._get_assert_msg(msg, debug_msg=debug_msg))
-            else:
-                result, debug_msg_generic = self._compareTensors(x, y, rtol=rtol, atol=atol,
-                                                                 equal_nan=equal_nan, exact_dtype=exact_dtype,
-                                                                 exact_device=exact_device)
-
-                if not result:
-                    assert debug_msg_generic is not None
-                    debug_msg = "Tensors failed to compare as equal!" + debug_msg_generic
-                super().assertTrue(result, msg=self._get_assert_msg(msg, debug_msg=debug_msg))
-        elif isinstance(x, (np.ndarray, torch.Tensor)) or isinstance(y, (np.ndarray, torch.Tensor)):
-            def maybe_to_tensor(a: Union[np.ndarray, torch.Tensor]) -> Union[np.ndarray, torch.Tensor]:
-                if not isinstance(a, np.ndarray):
-                    return a
 
+        if isinstance(x, torch.Tensor) and isinstance(y, torch.Tensor):
+            # In order to honor the @toleranceOverride and @precisionOverride decorators, we need to resolve the
+            # tolerances before we call assert_close
+            if rtol is None or atol is None:
+                rtol, atol = _get_default_rtol_and_atol(x, y)
+            rtol = max(rtol, self.rel_tol)
+            atol = max(atol, self.precision)
+
+            torch.testing.assert_close(
+                x,
+                y,
+                rtol=rtol,
+                atol=atol,
+                check_device=exact_device,
+                check_dtype=exact_dtype,
+                check_stride=exact_stride,
+                check_is_coalesced=exact_is_coalesced,
+                equal_nan=equal_nan,
+                msg=msg,
+            )
+        elif isinstance(x, np.ndarray) or isinstance(y, np.ndarray):
```

Comment:
Can this clause be eliminated just by checking if x or y are ndarrays at the start of the function and passing them to torch.from_numpy() if they are?

Reply:
Nope. I've hit a case where the test checked an `ndarray` without a numeric dtype. Thus, we are trying to convert it into a tensor and otherwise use a list as fallback.

Reply:
Holy cow. What test is that!?

Reply:
I need to investigate again. I'll get back to you.

Reply:
See https://app.circleci.com/pipelines/github/pytorch/pytorch/354868/workflows/510cc2a2-0d0f-4dfc-9c21-5a1cf97900d2/jobs/14957950 for an example.